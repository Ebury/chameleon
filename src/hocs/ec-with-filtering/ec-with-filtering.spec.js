import { mount } from '@vue/test-utils';
import { defineComponent, h } from 'vue';

import withFiltering from './ec-with-filtering';

describe('EcWithFiltering', () => {
  function mountEcWithFiltering(props, mountOpts) {
    const Component = defineComponent({
      props: ['filter'],
      render() {
        return h('div', {}, `Filter: ${JSON.stringify(this.filter)}`);
      },
    });

    const hocWrapper = mount(withFiltering(Component), {
      props,
      ...mountOpts,
    });

    const componentWrapper = hocWrapper.findComponent(Component);

    return { hocWrapper, componentWrapper };
  }

  it('should render properly', () => {
    const { hocWrapper } = mountEcWithFiltering();
    expect(hocWrapper.element).toMatchSnapshot();
  });

  it('should render the given filter', () => {
    const { hocWrapper } = mountEcWithFiltering({ filter: { testProp: 'Random value' } });
    expect(hocWrapper.element).toMatchSnapshot();
  });

  it('should update the filtering when filtering event is emitted', async () => {
    const { hocWrapper, componentWrapper } = mountEcWithFiltering();

    await componentWrapper.vm.$emit('filtering', { testProp: 'Random value' });
    expect(hocWrapper.element).toMatchSnapshot();
  });
});
