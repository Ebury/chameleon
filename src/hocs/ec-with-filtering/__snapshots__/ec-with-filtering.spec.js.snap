// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcWithFiltering should render properly 1`] = `
<div>
  Filter: {}
</div>
`;

exports[`EcWithFiltering should render the given filter 1`] = `
<div>
  Filter: {"testProp":"Random value"}
</div>
`;

exports[`EcWithFiltering should update the filtering when filtering event is emitted 1`] = `
<div>
  Filter: {"testProp":"Random value"}
</div>
`;
